azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/ExpressRouteCircuit.java

```diff
@@ -0,0 +1,223 @@
+
+    /**
+     * @return the ServiceKey
+     */
+    String serviceKey();
+
+    /**
+     * @return the ServiceProviderNotes
+     */
+    String serviceProviderNotes();
+
+    /**
+     * @return the ServiceProviderProperties
+     */
+    ExpressRouteCircuitServiceProviderProperties serviceProviderProperties();
+
+    /**
+     * Gets the provisioning state of the express route circuit resource. Possible values
+     * are: 'Updating', 'Deleting', and 'Failed'.
+     * @return provisioningState
+     */
+    String provisioningState();
+
+    /**
+     * The entirety of the express route circuit definition.
+     */
+    interface Definition extends
+            DefinitionStages.Blank,
+            DefinitionStages.WithGroup,
+            DefinitionStages.WithServiceProvider,
+            DefinitionStages.WithPeeringLocation,
+            DefinitionStages.WithBandwidth,
+            DefinitionStages.WithSkuTier,
+            DefinitionStages.WithSkuFamily,
+            DefinitionStages.WithCreate {
+    }
+
+    /**
+     * Grouping of express route circuit definition stages.
+     */
+    interface DefinitionStages {
+        /**
+         * The first stage of express route circuit definition.
+         */
+        interface Blank extends GroupableResource.DefinitionWithRegion<WithGroup> {
+        }
+
+        /**
+         * The stage of express route circuit definition allowing to specify the resource group.
+         */
+        interface WithGroup
+                extends GroupableResource.DefinitionStages.WithGroup<DefinitionStages.WithServiceProvider> {
+        }
+
+        /**
+         * The stage of express route circuit definition allowing to specify service provider name.
+         */
+        interface WithServiceProvider {
+            WithPeeringLocation withServiceProvidet(String serviceProviderName);
+        }
+
+        /**
+         * The stage of express route circuit definition allowing to specify service provider peering location.
+         */
+        interface WithPeeringLocation {
+            WithBandwidth withPeeringLocation(String location);
+        }
+
+        /**
+         * The stage of express route circuit definition allowing to specify service provider bandwidth.
+         */
+        interface WithBandwidth {
+            WithSkuTier withBandwidthInMbps(int bandwidthInMbps);
+        }
+
+        /**
+         * The stage of express route circuit definition allowing to specify SKU tier. Possible values include: 'Standard', 'Premium'.
+         */
+        interface WithSkuTier {
+            WithSkuFamily withSkuTier(ExpressRouteCircuitSkuTier skuTier);
+        }
+
+        /**
+         * The stage of express route circuit definition allowing to specify SKU family. Possible values include: 'UnlimitedData', 'MeteredData'.
+         */
+        interface WithSkuFamily {
+            WithCreate withSkuFamily(ExpressRouteCircuitSkuFamily skuFamily);
+        }
```

Comment:
If we define an enum `ExpressRouteCircuitSkuTypes` as mentioned in the previous comment then we can have a single wither `withSkuType(ExpressRouteCircuitSkuTypes sku)` that replaces `withSkuTier` and `WithSkuFamily`